tests/qdwh_test.py

```diff
@@ -103,30 +106,31 @@ def testQdwhWithUpperTriangularInputAllOnes(self, m, n, log_cond):
     rtol = 1E6 * _POLAR_TEST_EPS
 
     with self.subTest('Test u.'):
-      self.assertAllClose(actual_u, expected_u, rtol=rtol)
+      relative_diff_u = _compute_relative_diff(actual_u, expected_u)
+      np.testing.assert_almost_equal(relative_diff_u, 1E-4, decimal=2)
 
     with self.subTest('Test h.'):
-      self.assertAllClose(actual_h, expected_h, rtol=rtol)
+      relative_diff_h = _compute_relative_diff(actual_h, expected_h)
+      np.testing.assert_almost_equal(relative_diff_h, 1E-4, decimal=2)
 
     with self.subTest('Test u.dot(h).'):
       a_round_trip = actual_u.dot(actual_h)
```

Comment:
Is it possible that the TPU precision problems occur because JAX currently defaults to low precision for matmuls on TPU? I think calling `.dot` will do that.

Try using the scope:
```
with jax.default_matmul_precision('float32'):
```

or passing the `precision=` argument to methods like `dot`.